torch/_prims/nvfuser_executor.py

```diff
@@ -324,25 +324,18 @@ def _remove_empty_like_fill(gm: GraphModule):
     # This is a workaround for nonoptimal traces of C++ code `(1 - tensor)`
     # https://github.com/pytorch/pytorch/issues/86612
 
-    # Here when we see a `sub` node, we check if the first input is a result of
-    # filling a tensor with a scalar
-    # If so, we replace the first argument of the `sub` node with a scalar
-    for node in gm.graph.nodes:
-        if node.op == "call_function":
-            if node.target == torch.ops.nvprims.sub.default:
-                # check if the first argument is a fill
-                if (
-                    isinstance(node.args[0], torch.fx.Node)
-                    and node.args[0].op == "call_function"
-                    and node.args[0].target == torch.ops.aten.fill.Scalar
-                ):
-                    # Replace the first argument with the second argument of fill
-                    # aten.fill.Scalar(tensor, scalar)
-                    fill_node = node.args[0]
-                    scalar = fill_node.args[1]
-                    node.args = (scalar, *node.args[1:])
-    gm.graph.eliminate_dead_code()
-    gm.recompile()
+    def pattern(scalar, tensor):
+        empty_like = torch.ops.aten.empty_like.default(
```

Comment:
In the existing code, I was trying to match "empty_like+fill+sub" that's what is traced from the `scalar - tensor` code in C++. empty_like wasn't in the existing code because I thought it was not necessary. The hanging `empty_like` node was removed with `gm.graph.eliminate_dead_code()` call, now it's removed by matching the pattern.
Yes, the existing code is a bit more general, but that wasn't done intentionally.
I can remove `empty_like` from the pattern and bring back `gm.graph.eliminate_dead_code()`.

Reply:
Let's keep it there since it's a safer pattern to replace.

Meanwhile, let's put a comment there mentioning that the replaced pattern could be improved if we have utils to check the broadcasting semantics.